django/db/models/fields/related.py

```diff
@@ -1030,8 +1036,59 @@ def _check_on_delete(self):
                     id="fields.E321",
                 )
             ]
-        else:
-            return []
+        elif (
+            on_delete in [DB_CASCADE, DB_SET_NULL]
+            and hasattr(self.model, "_meta")
+            and any(
+                not parent._meta.abstract
+                for parent in self.model._meta.get_parent_list()
```

Comment:
It might be helpful to specify which parent model(s) (or `parent_ptr` field(s)) are incompatible.

Reply:
Then we have to move to a for loop instead of the `any()`